nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/renditions/RenditionsObject.java

```diff
@@ -0,0 +1,147 @@
+import org.nuxeo.runtime.kv.KeyValueStore;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "renditions")
+public class RenditionsObject extends DefaultObject {
+
+    protected Set<State> notAtWork = EnumSet.of(State.UNKNOWN, State.ABORTED, State.COMPLETED);
+
+    protected Lock picturesLock = new ReentrantLock();
+
+    protected Lock thumbnailsLock = new ReentrantLock();
+
+    public static final String RECOMPUTE_VIEWS_ACTION_COMMAND_ID = RecomputeViewsAction.ACTION_NAME + "_commandId";
+
+    public static final String RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID = RecomputeThumbnailsAction.ACTION_NAME
+            + "_commandId";
+
+    public static final String THUMBNAILS_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Thumbnail' AND thumb:thumbnail/data IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0";
+
+    public static final String PICTURES_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Picture' AND picture:views/*/title IS NULL";
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doPostPictures() throws OperationException {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String commandId = getCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID);
+        if (picturesLock.tryLock()) {
+            try {
+                if (needToRun(bulkService, commandId)) {
+                    commandId = bulkService.submit(new BulkCommand.Builder(RecomputeViewsAction.ACTION_NAME,
+                            PICTURES_DEFAULT_QUERY).user(ctx.getPrincipal().getName())
+                                                   .param(RecomputeViewsAction.PARAM_XPATH, "file:content")
+                                                   .build());
+                    putCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID, commandId);
+                }
+            } finally {
+                picturesLock.unlock();
+            }
+        }
+        return bulkService.getStatus(commandId);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doGetPictures() {
+        BulkService service = Framework.getService(BulkService.class);
+        return service.getStatus(getCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID));
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("thumbnails/recompute")
+    public BulkStatus doPostThumbnails() throws OperationException {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String commandId = getCommandId(RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID);
+        if (thumbnailsLock.tryLock()) {
+            try {
+                if (needToRun(bulkService, commandId)) {
+                    commandId = bulkService.submit(new BulkCommand.Builder(RecomputeThumbnailsAction.ACTION_NAME,
+                            THUMBNAILS_DEFAULT_QUERY).user(ctx.getPrincipal().getName()).build());
```

Comment:
You mean instead of the `NuxeoPrincipal#getName` ?

Reply:
Indeed

Reply:
as discussed we will replace / provide the system user instead of the context user, because it will be cases where the not system user will not be defined in Nuxeo. Actually Nuxeo 10.10 doesn't handle the System user in BAF ([here](https://github.com/nuxeo/nuxeo/blob/b80483241d954284c48a027d4ca59032d0fa519f/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/computation/BulkScrollerComputation.java#L126) but i's the case on master ([here](https://github.com/nuxeo/nuxeo/blob/4f20846436c1926d6ad04bc7b02dd777173ef699/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/computation/BulkScrollerComputation.java#L128))

see with efge the jira https://jira.nuxeo.com/browse/NXP-27494 will be backported in part on 10.10 by his devs on retention. After these backport we can easy just provide the System user.